test/beast/websocket/read3.cpp

```diff
@@ -625,170 +624,369 @@ class read3_test : public websocket_test_suite
+                "\x9d\x4c\xa3\xe3\xbe\xf4\x45\x5c\x95\xde\xa5"
+                "\x1c\xcd\x8a\x21\xb7\x12\x13\x89\x99\x84\x42"
+                "\x9e\x22\x39\x98\x1b\x19\x67\x4e\x4b\xf0\x95"
+                "\x8c\x20\x51\x40\x47\xdf\x9b\x7e\x8f\xab\xd2"
+                "\x8f\x28\x50\xc6\x4c\x4c\xc9\xb1\x25\x81\x99"
+                "\xa1\xed\x66\x84\x5c\xe4\x24\x78\x10\x4e\x59"
+                "\x4a\x19\xd4\x35\x73\x82\xdd\xea\x9c\xfb\x22"
+                "\xae\x7e\x7b\x37\xa1\x13\xf2\x91\x4c\x90\x13"
+                "\xd6\x9f\x77\x14\xd3\x6c\xb0\x14\x04\xb6\x0e"
+                "\x7b\x97\x49\xb5\x0c\xa3\x8f\x32\xe2\x7b\xdf"
+                "\xfe\x12\x57\xa5\x4f\x92\x38\x25\x83\xa5\x0e"
+                "\x1b\x47\x02\x37\x49\xe8\xeb\x85\x22\x2c\x50"
+                "\xf2\x28\xf9\x6b\x0e\x85\x0c\xf6\x24\xf6\x58"
+                "\xaa\x63\x5f\xf6\x4b\x58\x95\xdc\x66\xb1\x71"
+                "\x11\x64\x83\x7d\x20\x58\x87\x89\xe2\xcc\x81"
+                "\xe0\x5d\xe3\x92\x45\xd8\x2c\x5c\x81\x6c\xed"
+                "\x10\xdb\xb5\x9a\xf2\x57\x01\xb2\xdf\x54\x91"
+                "\x9b\x92\x72\xed\x08\x5a\x3f\xe2\x76\x07\x45"
+                "\xac\xa0\xd6\xb6\x02\xf6\xfc\xf9\xe9\xfb\x5d"
+                "\x16\xb1\xa1\xf4\xf6\xac\xd4\x9a\x52\x6b\x4a"
+                "\xad\x29\xb5\xa6\xd4\x9a\x52\x6b\x4a\xad\x29"
+                "\xb5\xa6\xd4\x9a\x52\x6b\x4a\xad\x29\xb5\xa6"
+                "\xd4\x9a\x52\x6b\x4a\xad\x29\xb5\xa6\xd4\x9a"
+                "\x52\x6b\x4a\xad\x29\xb5\xa6\xd4\x9a\x52\x6b"
+                "\x4a\xad\x29\xb5\xa6\xd4\x9a\x52\x6b\xff\x27"
+                "\xb5\x66\xab\xb5\x7f\x7e\xad\x96\x1b\xe2\xa1"
+                "\xe3\x7e\x63\x1b\xb0\x9f\x36\xf8\x98\x79\x18"
+                "\x3a\x3e\x92\x5d\xb8\x05\x27\x28\x84\x0f\x4f"
+                "\xab\x77\xe1\x16\x50\xa0\x10\x3e\xec\xd6\x2e"
+                "\xc2\x0d\xfc\xc5\xd0\xd1\x06\xec\xc2\x2d\x54"
+                "\x4d\x21\x7c\x78\xbc\xb0\x0b\xb7\x40\x0b\x85"
+                "\xf0\xe1\x89\xe0\x2e\xdc\x02\xe2\x14\xc2\x87"
+                "\xe7\xe9\xef\xcd\xf0\xe5\x2f\x14\xe2\x0f\x00"
+            ),
+
+            // packet 4
+            sbuf(
+                "\xc1\x45"
+                "\xec\xda\x31\x0d\x00\x00\x00\xc3\x20\xff\xae"
+                "\x67\xa1\xf7\x82\x0e\x68\x0b\x91\x1a\x53\x63"
+                "\x6a\x4c\x8d\xa9\x31\x35\xa6\xc6\xd4\x98\x1a"
+                "\x53\x63\x6a\x4c\x8d\xa9\x31\x35\xa6\xc6\xd4"
+                "\x98\x1a\x53\x63\x6a\x4c\x8d\xa9\x31\x35\xa6"
+                "\xc6\xd4\x98\x1a\x53\x63\x6a\x4c\x8d\xa9\xf1"
+                "\xa7\x1a\x0f"
+            ),
         };
 
-        for (const auto & line : lines) {
-            std::vector<uint8_t> out;
-            boost::algorithm::unhex(line.begin(), line.end(), std::back_inserter(out));
-            net::write(wss.next_layer(), net::const_buffer(out.data(), out.size()));
+        for (const auto & packet : packets) {
```

Comment:
I would have used `const packet` instead of a reference but I suppose it doesn't matter